src/nnn.c

```
@@ -4425,92 +4425,106 @@ static void set_smart_ctx(int ctx, char *nextpath, char **path, char *file, char
 }
 
 /*
- * Gets only a single line (that's what we need for now) or shows full command output in pager.
- * Uses g_buf internally.
+ * This function does one of the following depending on the values of `fdout` and `page`:
+ *  1) fdout == -1 && !page: Write up to CMD_LEN_MAX bytes of command output into g_buf
+ *  2) fdout == -1 && page: Create a temp file, write full command output into it and show in pager.
+ *  3) fdout != -1 && !page: Write full command output into the provided file.
+ *  4) fdout != -1 && page: Don't use! Returns FASLE.
+ *
+ * g_buf is modified only in case 1.
+ * g_tmpfpath is modified only in case 2.
  */
-static bool get_output(char *file, char *arg1, char *arg2, int fdout, bool multi, bool page)
+static bool get_output(char *file, char *arg1, char *arg2, int fdout, bool page)
 {
 	pid_t pid;
 	int pipefd[2];
 	int index = 0, flags;
 	bool ret = FALSE;
-	bool tmpfile = ((fdout == -1) && page);
-	char *argv[EXEC_ARGS_MAX] = {0};
-	char *cmd = NULL;
-	int fd = -1;
+	bool have_file = fdout != -1;
+	int cmd_in_fd = -1;
+	int cmd_out_fd = -1;
 	ssize_t len;
 
-	if (tmpfile) {
+	/*
+	 * In this case the logic of the function dictates that we should write the output of the command
+	 * to `fd` and show it in the pager. But since we didn't open the file descriptor we have no right
+	 * to close it, the caller must do it. We don't even know the path to pass to the pager and
+	 * it's a real hassle to get it. In general this just invites problems so we are blocking it.
+	 */
+	if (have_file && page)
+		return FALSE;
```

Is there any actual cases of this function being called like this? I'd personally use `assert` but `nnn` codebase doesn't seem to use it. A debug log would be nice I think.

---

No, there are currently no code paths that do this, but I thought I'd still prevent it from going through. An `assert` is a bit aggressive. We don't need to crash the whole program, just prevent shenanigans from happening. I'll add a debug log, good idea.

---

> An `assert` is a bit aggressive. We don't need to crash the whole program, just prevent shenanigans from happening.

The idea is that if you compile with `-N DEBUG` then the "release build" won't suffer from it and in "debug builds" it will get caught easily and instantly (instead of potentially silently ignoring the bogus argument and having the program march on forward as if everything is ok).

But a debug log is fine by me too, especially since `nnn` doesn't use any assert currently.